simple-mbed-cloud-client.cpp

```diff
@@ -267,29 +304,42 @@ bool SimpleMbedCloudClient::is_register_called() {
     return _register_called;
 }
 
-void SimpleMbedCloudClient::register_and_connect() {
-    // TODO this might not work if called more than once...
+bool SimpleMbedCloudClient::register_and_connect() {
+    if (_register_and_connect_called) return false;
+
     mcc_resource_def resourceDef;
 
     // TODO clean up
-    for (unsigned int i = 0; i < _resources.size(); i++) {
+    for (int i = 0; i < _resources.size(); i++) {
         _resources[i]->get_data(&resourceDef);
         M2MResource *res = add_resource(&_obj_list, resourceDef.object_id, resourceDef.instance_id,
-                     resourceDef.resource_id, resourceDef.name.c_str(), M2MResourceInstance::STRING,
-                     (M2MBase::Operation)resourceDef.method_mask, resourceDef.value.c_str(), resourceDef.observable,
-                     resourceDef.put_callback, resourceDef.post_callback, resourceDef.notification_callback);
-        _resources[i]->set_resource(res);
+                    resourceDef.resource_id, resourceDef.name.c_str(), M2MResourceInstance::STRING,
+                    (M2MBase::Operation)resourceDef.method_mask, resourceDef.value.c_str(), resourceDef.observable,
+                    resourceDef.put_callback, resourceDef.post_callback, resourceDef.notification_callback);
+        _resources[i]->set_m2m_resource(res);
     }
     _cloud_client.add_objects(_obj_list);
 
+    _register_and_connect_called = true;
```

Comment:
I doesn't look like we're setting this back to `false` when we deregister. Is the intent that this function should only be called once?

Reply:
Yes, as it adds new resources. You can call `call_register()` when doing it the second time. It's public and does not have this guard.